### ISSUE TITLE
[java] InvalidLogMessageFormat: false positives with logstash structured logging

### ISSUE THREAD

**Issue by Alexx-G** [pmd/pmd#3644]
**Affects PMD Version:** 6.36.0

**Rule:** [InvalidLogMessageFormat](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#invalidlogmessageformat)

**Description:**

**Code Sample demonstrating the issue:**

When using https://github.com/logfellow/logstash-logback-encoder for structured logging (example below), PMD detects it as false positive.

With markers:
```java
import static net.logstash.logback.marker.Markers.*;

// Add "foo:bar" to the JSON output
    LOGGER.info(append("foo", "bar"), "Some log message");
// Add "foo:bar,name:value" to the JSON output
    LOGGER.info(append("foo", "bar").and(append("name", "value")), "Some log message");
```

With structured arguments:
```java
import static net.logstash.logback.argument.StructuredArguments.*;

// Add "foo:bar" to JSON output and "bar" to log message
    LOGGER.info("log message {}", value("foo", "bar"));
// Add "foo:bar" to JSON output only
    LOGGER.info("log message", keyValue("foo", "bar"));
```

**Expected outcome:**

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*
